### DISCUSSION TITLE
Move invite functionality into separate module

### DISCUSSION THREAD

**Discussion by MekDrop** [ImpressCMS/impresscms#1047] [Ideas]
And yes this module should be suggested by composer!

Less not frequently used code in core = easier management.

**Comment by fiammybe** [comment]
The entire user handling needs reviewing. This atomically splitting of micro-functionalities is more the way Drupal and Wordpress handle stuff, so we should take care with how we handle that. I've been quite frustrated with Drupal when I needed some functionality, but that was is a small package, depending on other packages that had conflicting version requirements. Nightmare :-) 

If you want to move this into a separate package, I think you will need to have some events defined that will let you plug into the basic icms flows of registering a user, user-related information tables and so on.

**Comment by MekDrop** [comment]
I think invites doesn't depend on user heavily. It can be separated. 

From otherside not many people knows how this functionality works I think. I didn't see anything that used this functionality. So, we could reimagine it.

**Reply by fiammybe** [reply]
I haven't used it in all my years with ImpressCMS, so yes : reimagine and get it out of the core :-)
